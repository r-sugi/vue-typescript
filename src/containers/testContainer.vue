<template>
  <div>
    <testTable
      :headers="headers"
      :items="items"
      :items-per-page="itemsPerPage"
      :sort-by="sortBy"
      :sort-desc="sortDesc"
      @changeItemsPerPage="changeItemsPerPage"
    >
    </testTable>
  </div>
</template>
<script>
import testTable from "@/components/testTable";
export default {
  components: {
    testTable
  },
  data() {
    return {
      itemsPerPage: 2,
      //TODO: change header based on page url
      headers: [
        {
          text: "Dessert (100g serving)",
          align: "start",
          value: "name"
        },
        { text: "Fat (g)", value: "fat" }
      ],
      // items from api
      items: [
        {
          name: "Frozen Yogurt",
          fat: 6.0
        },
        {
          name: "Ice cream sandwich",
          fat: 9.0
        },
        {
          name: "Eclair",
          fat: 16.0
        },
        {
          name: "Cupcake",
          fat: 3.7
        }
      ],
      // control by container
      sortBy: ["name"],
      // control by container
      sortDesc: [false]
    };
  },
  methods: {
    changeItemsPerPage(num) {
      this.itemsPerPage = num;
    }
  }
};
</script>
