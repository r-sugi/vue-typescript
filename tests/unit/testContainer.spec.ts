import { mount } from "@vue/test-utils";
import vuetify from "@/plugins/vuetify.js";
import testContainer from "@/containers/testContainer.vue";
(window as any).globalThis = window;
globalThis.innerWidth = 1200;

describe("WIP", () => {
  it("should render", async () => {
    const wrapper = mount(testContainer, {
      vuetify
    });
    expect(true).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
