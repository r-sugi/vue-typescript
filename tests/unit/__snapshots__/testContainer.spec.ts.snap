// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WIP should render 1`] = `
<div>
  <div
    class="v-data-table theme--light"
  >
    <div
      class="v-data-table__wrapper"
    >
      <table>
        <colgroup>
          <col
            class=""
          />
          <col
            class=""
          />
          <col
            class=""
          />
          <col
            class=""
          />
          <col
            class=""
          />
          <col
            class=""
          />
        </colgroup>
        <thead
          class="v-data-table-header"
        >
          <tr>
            <th
              aria-label="Dessert (100g serving)"
              class="text-start"
              role="columnheader"
              scope="col"
            >
              <span>
                Dessert (100g serving)
              </span>
            </th>
            <th
              aria-label="Calories: Not sorted. Activate to sort ascending."
              aria-sort="none"
              class="text-start sortable"
              role="columnheader"
              scope="col"
            >
              <span>
                Calories
              </span>
              <i
                aria-hidden="true"
                class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light"
                style="font-size: 18px;"
              />
            </th>
            <th
              aria-label="Fat (g): Not sorted. Activate to sort ascending."
              aria-sort="none"
              class="text-start sortable"
              role="columnheader"
              scope="col"
            >
              <span>
                Fat (g)
              </span>
              <i
                aria-hidden="true"
                class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light"
                style="font-size: 18px;"
              />
            </th>
            <th
              aria-label="Carbs (g): Not sorted. Activate to sort ascending."
              aria-sort="none"
              class="text-start sortable"
              role="columnheader"
              scope="col"
            >
              <span>
                Carbs (g)
              </span>
              <i
                aria-hidden="true"
                class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light"
                style="font-size: 18px;"
              />
            </th>
            <th
              aria-label="Protein (g): Not sorted. Activate to sort ascending."
              aria-sort="none"
              class="text-start sortable"
              role="columnheader"
              scope="col"
            >
              <span>
                Protein (g)
              </span>
              <i
                aria-hidden="true"
                class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light"
                style="font-size: 18px;"
              />
            </th>
            <th
              aria-label="Iron (%): Not sorted. Activate to sort ascending."
              aria-sort="none"
              class="text-start sortable"
              role="columnheader"
              scope="col"
            >
              <span>
                Iron (%)
              </span>
              <i
                aria-hidden="true"
                class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light"
                style="font-size: 18px;"
              />
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              Frozen Yogurt
            </td>
            <td
              class="text-start"
            >
              159
            </td>
            <td
              class="text-start"
            >
              6
            </td>
            <td
              class="text-start"
            >
              24
            </td>
            <td
              class="text-start"
            >
              4
            </td>
            <td
              class="text-start"
            >
              1%
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              Ice cream sandwich
            </td>
            <td
              class="text-start"
            >
              237
            </td>
            <td
              class="text-start"
            >
              9
            </td>
            <td
              class="text-start"
            >
              37
            </td>
            <td
              class="text-start"
            >
              4.3
            </td>
            <td
              class="text-start"
            >
              1%
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              Eclair
            </td>
            <td
              class="text-start"
            >
              262
            </td>
            <td
              class="text-start"
            >
              16
            </td>
            <td
              class="text-start"
            >
              23
            </td>
            <td
              class="text-start"
            >
              6
            </td>
            <td
              class="text-start"
            >
              7%
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              Cupcake
            </td>
            <td
              class="text-start"
            >
              305
            </td>
            <td
              class="text-start"
            >
              3.7
            </td>
            <td
              class="text-start"
            >
              67
            </td>
            <td
              class="text-start"
            >
              4.3
            </td>
            <td
              class="text-start"
            >
              8%
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              Gingerbread
            </td>
            <td
              class="text-start"
            >
              356
            </td>
            <td
              class="text-start"
            >
              16
            </td>
            <td
              class="text-start"
            >
              49
            </td>
            <td
              class="text-start"
            >
              3.9
            </td>
            <td
              class="text-start"
            >
              16%
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              Jelly bean
            </td>
            <td
              class="text-start"
            >
              375
            </td>
            <td
              class="text-start"
            >
              0
            </td>
            <td
              class="text-start"
            >
              94
            </td>
            <td
              class="text-start"
            >
              0
            </td>
            <td
              class="text-start"
            >
              0%
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              Lollipop
            </td>
            <td
              class="text-start"
            >
              392
            </td>
            <td
              class="text-start"
            >
              0.2
            </td>
            <td
              class="text-start"
            >
              98
            </td>
            <td
              class="text-start"
            >
              0
            </td>
            <td
              class="text-start"
            >
              2%
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              Honeycomb
            </td>
            <td
              class="text-start"
            >
              408
            </td>
            <td
              class="text-start"
            >
              3.2
            </td>
            <td
              class="text-start"
            >
              87
            </td>
            <td
              class="text-start"
            >
              6.5
            </td>
            <td
              class="text-start"
            >
              45%
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              Donut
            </td>
            <td
              class="text-start"
            >
              452
            </td>
            <td
              class="text-start"
            >
              25
            </td>
            <td
              class="text-start"
            >
              51
            </td>
            <td
              class="text-start"
            >
              4.9
            </td>
            <td
              class="text-start"
            >
              22%
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="text-start"
            >
              KitKat
            </td>
            <td
              class="text-start"
            >
              518
            </td>
            <td
              class="text-start"
            >
              26
            </td>
            <td
              class="text-start"
            >
              65
            </td>
            <td
              class="text-start"
            >
              7
            </td>
            <td
              class="text-start"
            >
              6%
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="v-data-footer"
    >
      <div
        class="v-data-footer__select"
      >
        Rows per page:
        <div
          class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select"
        >
          <div
            class="v-input__control"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-owns="list-13"
              class="v-input__slot"
              role="button"
            >
              <div
                class="v-select__slot"
              >
                <div
                  class="v-select__selections"
                >
                  <div
                    class="v-select__selection v-select__selection--comma"
                  >
                    5
                  </div>
                  <input
                    aria-label="$vuetify.dataTable.itemsPerPageText"
                    aria-readonly="false"
                    autocomplete="off"
                    id="input-13"
                    readonly="readonly"
                    type="text"
                  />
                </div>
                <div
                  class="v-input__append-inner"
                >
                  <div
                    class="v-input__icon v-input__icon--append"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon notranslate mdi mdi-menu-down theme--light"
                    />
                  </div>
                </div>
                <input
                  type="hidden"
                  value="[object Object]"
                />
              </div>
              <div
                class="v-menu"
              >
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="v-data-footer__pagination"
      >
        1-10 of 10
      </div>
      <div
        class="v-data-footer__icons-before"
      >
        <button
          aria-label="Previous page"
          class="v-btn v-btn--disabled v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default"
          disabled="disabled"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-chevron-left theme--light"
            />
          </span>
        </button>
      </div>
      <div
        class="v-data-footer__icons-after"
      >
        <button
          aria-label="Next page"
          class="v-btn v-btn--disabled v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default"
          disabled="disabled"
          type="button"
        >
          <span
            class="v-btn__content"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-chevron-right theme--light"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
